src/Simplex/Chat.hs

```diff
@@ -3542,13 +3542,23 @@ processAgentMessageConn user@User {userId} corrId agentConnId agentMessage = do
           let ExtMsgContent _ _ itemTTL live_ = mcExtMsgContent mc
               timed_ = rcvGroupCITimed gInfo itemTTL
               live = fromMaybe False live_
-          file_ <- processFileInvitation fInv_ content $ \db -> createRcvGroupFileTransfer db userId m
-          ChatItem {formattedText} <- newChatItem (CIRcvMsgContent content) (snd <$> file_) timed_ live
-          autoAcceptFile file_
-          let g = groupName' gInfo
-          whenGroupNtfs user gInfo $ do
-            showMsgToast ("#" <> g <> " " <> c <> "> ") content formattedText
-            setActive $ ActiveG g
+          -- check if message moderation event was received ahead of message
+          withStore' (\db -> findModeration db user gInfo memberId sharedMsgId_) >>= \case
+            Nothing -> do
+              file_ <- processFileInvitation fInv_ content $ \db -> createRcvGroupFileTransfer db userId m
+              ChatItem {formattedText} <- newChatItem (CIRcvMsgContent content) (snd <$> file_) timed_ live
+              autoAcceptFile file_
+              let g = groupName' gInfo
+              whenGroupNtfs user gInfo $ do
+                showMsgToast ("#" <> g <> " " <> c <> "> ") content formattedText
+                setActive $ ActiveG g
+            Just (moderationId, moderatorMember, createdByMsgId, moderatedAtTs) -> do
+              unless (groupFeatureAllowed SGFFullDelete gInfo) $ do
+                file_ <- processFileInvitation fInv_ content $ \db -> createRcvGroupFileTransfer db userId m
+                ci <- newChatItem (CIRcvMsgContent content) (snd <$> file_) timed_ False
+                cr <- markGroupCIDeleted user gInfo (CChatItem SMDRcv ci) createdByMsgId False (Just moderatorMember) moderatedAtTs
+                toView cr
+              withStore' (`deleteChatItemModeration` moderationId)
```

Comment:
why do we need to delete it here? maybe it's ok...

Reply:
to clean up, why keep it after it's been applied